spec/rubocop/cop/rspec/expect_change_spec.rb

```diff
@@ -67,17 +67,47 @@
       RUBY
     end
 
+    it 'flags change matcher without block with nested class' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { run }.to change(User::Token, :count).by(1)
+                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `change { User::Token.count }`.
+        end
+      RUBY
+
+      expect_correction(<<-RUBY)
+        it do
+          expect { run }.to change { User::Token.count }.by(1)
+        end
+      RUBY
+    end
+
+    it 'flags change matcher without block with root class' do
+      expect_offense(<<-RUBY)
+        it do
+          expect { run }.to change(::Token, :count).by(1)
+                            ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `change { ::Token.count }`.
+        end
+      RUBY
+
+      expect_correction(<<-RUBY)
+        it do
+          expect { run }.to change { ::Token.count }.by(1)
+        end
+      RUBY
+    end
+
     it 'flags change matcher when receiver is a variable' do
       expect_offense(<<-RUBY)
         it do
-          expect { run }.to change(User, :count)
-                            ^^^^^^^^^^^^^^^^^^^^ Prefer `change { User.count }`.
+          expect { run }.to change(user, :count)
```

Comment:
I'm not sure I completely understand the reason behind this change `User` -> `user`.
`User.count` resembles Rails specs.

Reply:
It's very interesting, actually.

It seems that currently we only match consts correctly.
While your example is actually totally legit, e.g.
```ruby
expect { file.upload! }.to change(user.uploads, :count)
```

Do you mind to add something like this as a separate example for both styles?

Reply:
Thank you for your comments, (M-Yamashita is my colleague :) )

When `RSpec/ExpectChange` was introduced, this spec was `change(user, :count)` https://github.com/rubocop-hq/rubocop-rspec/commit/fc57a5cb7537b21e9e8457f419a4cdfbb80aa1e5

And this commit https://github.com/rubocop-hq/rubocop-rspec/commit/971fc7d38d9f4b5ea87b179fdf8c5d0b9c4b85b0 changed `user` to `User`.
However `change(User, :count)` is already tested in another spec ( `it 'flags change matcher without block'` ).
So we thought this is unexpected change and just reverted it (User to user).

> It seems that currently we only match consts correctly.
> While your example is actually totally legit, e.g.
>
> expect { file.upload! }.to change(user.uploads, :count)
> Do you mind to add something like this as a separate example for both styles?

We didn't noticed this pattern, but it's really interesting.
We will try it.

Reply:
I'm sorry for the late reply.
We added examples for method_call and block styles.